modules/mobile-agents/android/client/plugins/IDPProxy/src/org/wso2/mdm/agent/proxy/utils/ServerUtilities.java

```
@@ -0,0 +1,375 @@
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.DefaultHttpClient;
+import org.apache.http.impl.conn.tsccm.ThreadSafeClientConnManager;
+import org.apache.http.params.BasicHttpParams;
+import org.apache.http.params.HttpParams;
+import org.apache.http.protocol.HTTP;
+import org.json.JSONObject;
+import org.wso2.mdm.agent.proxy.IDPTokenManagerException;
+import org.wso2.mdm.agent.proxy.IdentityProxy;
+import org.wso2.mdm.agent.proxy.R;
+import org.wso2.mdm.agent.proxy.beans.APIUtilities;
+
+import java.io.*;
+import java.security.*;
+import java.security.cert.CertificateException;
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.*;
+import java.util.Map.Entry;
+
+/**
+ * Handle network communication between SDK and authorization server
+ */
+public class ServerUtilities {
+	private final static String TAG = "ServerUtilities";
+	private static final DateFormat dateFormat = new SimpleDateFormat("MM/dd/yyyy HH:mm:ss",
+	                                                                  Locale.getDefault());
+
+	/**
+	 * Validate the token expiration date.
+	 *
+	 * @param expirationDate - Token expiration date.
+	 * @return - Token status.
+	 */
+	public static boolean isValid(Date expirationDate) {
+		Date currentDate = new Date();
+		String formattedDate = dateFormat.format(currentDate);
+		currentDate = convertDate(formattedDate);
+
+		boolean isExpired = currentDate.after(expirationDate);
+		boolean isEqual = currentDate.equals(expirationDate);
+		if (isExpired == true || isEqual == true) {
+			return true;
+		}
+
+		return false;
+	}
+
+	/**
+	 * Convert the date to the standard format.
+	 *
+	 * @param date - Date as a string.
+	 * @return - Formatted date.
+	 */
+	public static Date convertDate(String date) {
+		Date receivedDate = null;
+		try {
+			receivedDate = dateFormat.parse(date);
+		} catch (ParseException e) {
+			Log.e(TAG, "Invalid date format." + e);
```

We need to throw the exception up in this particular line.